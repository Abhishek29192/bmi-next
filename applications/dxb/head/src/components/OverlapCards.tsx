import { Container, CTACard, Grid } from "@bmi-digital/components";
import ButtonBase, { ButtonBaseProps } from "@mui/material/ButtonBase";
import { graphql } from "gatsby";
import React from "react";
import withGTM from "../utils/google-tag-manager";
import Image from "./Image";
import { getCTA } from "./Link";
import { Data as PageInfoData } from "./PageInfo";
import { Data as PromoData } from "./Promo";
import { useSiteContext } from "./Site";
import styles from "./styles/OverlapCards.module.scss";
import Video from "./Video";

type Card =
  | Pick<
      PromoData,
      "__typename" | "title" | "cta" | "featuredMedia" | "featuredVideo"
    >
  | Pick<
      PageInfoData,
      "__typename" | "title" | "path" | "featuredMedia" | "featuredVideo"
    >;

// NOTE: Minimum two cards required.
export type Data = [Card, Card, ...Card[]];

const IntegratedOverlapCards = ({ data }: { data?: Data }) => {
  const { countryCode } = useSiteContext();

  const GTMButton = withGTM<ButtonBaseProps>(ButtonBase, { action: "to" });

  return (
    <div className={styles["OverlapCards"]} data-testid="overlap-cards-wrapper">
      <Container>
        <Grid spacing={3} container justifyContent="center">
          {data.map(({ title, featuredMedia, featuredVideo, ...rest }, key) => {
            const cta = getCTA(rest, countryCode);
            return (
              <Grid key={key} xs={12} sm={6} md={5} lg={3}>
                <CTACard
                  title={title}
                  buttonComponent={(props: ButtonBaseProps) => (
                    <GTMButton
                      gtm={{
                        id: "cta-click1",
                        label: title
                      }}
                      {...props}
                    />
                  )}
                  media={
                    featuredVideo ? (
                      <Video
                        {...featuredVideo}
                        data-testid={"overlap-cards-video"}
                      />
                    ) : (
                      <Image
                        {...featuredMedia}
                        data-testid={"overlap-cards-image"}
                      />
                    )
                  }
                  clickableArea={featuredVideo ? "heading" : "full"}
                  action={cta?.action}
                  data-testid="overlap-card"
                />
              </Grid>
            );
          })}
        </Grid>
      </Container>
    </div>
  );
};

export default IntegratedOverlapCards;

export const query = graphql`
  fragment OverlapCardFragment on ContentfulPromoOrPage {
    ...PromoCardFragment
    ...PageInfoCardFragment
  }
`;
