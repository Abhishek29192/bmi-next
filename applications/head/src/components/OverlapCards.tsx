import React, { useContext } from "react";
import { graphql } from "gatsby";
import Grid from "@bmi/grid";
import CTACard from "@bmi/cta-card";
import Container from "@bmi/container";
import styles from "./styles/OverlapCards.module.scss";
import { Data as PromoData } from "./Promo";
import { Data as PageInfoData } from "./PageInfo";
import { getCTA } from "./Link";
import { SiteContext } from "./Site";

type Card = (
  | Pick<PromoData, "__typename" | "title" | "cta">
  | Pick<PageInfoData, "__typename" | "title" | "path">
) & {
  featuredImage: {
    resized: {
      src: string;
    };
  } | null;
};

// NOTE: Minimum two cards required.
export type Data = [Card, Card, ...Card[]];

const IntegratedOverlapCards = ({ data }: { data?: Data }) => {
  const { countryCode } = useContext(SiteContext);

  return (
    <div className={styles["OverlapCards"]}>
      <Container>
        <Grid spacing={3} container justify="center">
          {data.map(({ title, featuredImage, ...rest }, key) => {
            const { action } = getCTA(rest, countryCode);

            return (
              <Grid item key={key} xs={12} sm={6} md={5} lg={3}>
                <CTACard
                  title={title}
                  imageSource={featuredImage?.resized?.src}
                  action={action}
                />
              </Grid>
            );
          })}
        </Grid>
      </Container>
    </div>
  );
};

export default IntegratedOverlapCards;

export const query = graphql`
  fragment OverlapCardFragment on ContentfulPromoOrPage {
    ... on ContentfulPromo {
      ...PromoFragment
      featuredImage {
        resized: resize(width: 684, toFormat: WEBP, jpegProgressive: false) {
          src
        }
      }
    }
    ... on ContentfulPage {
      title
      path
      featuredImage {
        resized: resize(width: 684, toFormat: WEBP, jpegProgressive: false) {
          src
        }
      }
    }
  }
`;
