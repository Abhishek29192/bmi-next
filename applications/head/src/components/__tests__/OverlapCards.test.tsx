import React from "react";
import OverlapCards, { Data } from "../OverlapCards";
import { render } from "@testing-library/react";
import mockConsole from "jest-mock-console";

beforeAll(() => {
  mockConsole();
});

describe("OverlapCards component", () => {
  it("renders correctly", () => {
    const data: Data = [
      {
        title: "Call to action",
        slug: "some-page",
        featuredImage: {
          resize: {
            src: "link-to-page.png"
          }
        }
      },
      {
        title: "Call to action",
        slug: "some-page",
        featuredImage: {
          resize: {
            src: "link-to-page.png"
          }
        }
      }
    ];

    const { container } = render(<OverlapCards data={data} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
