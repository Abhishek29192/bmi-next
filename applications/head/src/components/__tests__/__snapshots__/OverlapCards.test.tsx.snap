// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OverlapCards component renders correctly 1`] = `
<div
  class="OverlapCards"
>
  <div
    class="Container"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthXl"
    >
      <div
        class="MuiGrid-root Grid Grid--justify MuiGrid-container MuiGrid-spacing-xs-3 MuiGrid-justify-xs-center"
      >
        <div
          class="MuiGrid-root item MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-5 MuiGrid-grid-lg-3"
        >
          <a
            class="MuiButtonBase-root Card"
            href="/uk/some-page"
          >
            <div
              class="MuiPaper-root MuiCard-root body MuiPaper-elevation1 MuiPaper-rounded"
            >
              <section
                class="top-box"
              >
                <h5
                  class="MuiTypography-root heading Typography MuiTypography-h5"
                >
                  Call to action
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root icon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
                    />
                  </svg>
                </h5>
              </section>
              <div
                class="image Media gatsby-image-wrapper"
                style="overflow: hidden;"
              >
                <div
                  aria-hidden="true"
                  style="width: 100%; padding-bottom: 100%;"
                />
                <picture>
                  
                  <img
                    alt="Lorem ipsum"
                    draggable="false"
                    loading="lazy"
                    sizes=""
                    src=""
                    srcset=""
                    style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 0; transition: opacity 500ms; font-family: \\"object-fit: cover; object-position: center\\";"
                  />
                </picture>
                <noscript>
                  &lt;picture&gt;&lt;source srcset="" /&gt;&lt;img loading="lazy" src="" alt="Lorem ipsum" style="position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
                </noscript>
              </div>
            </div>
            <span
              class="MuiTouchRipple-root"
            />
          </a>
        </div>
        <div
          class="MuiGrid-root item MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-5 MuiGrid-grid-lg-3"
        >
          <a
            class="MuiButtonBase-root Card"
            href="/uk/some-page"
          >
            <div
              class="MuiPaper-root MuiCard-root body MuiPaper-elevation1 MuiPaper-rounded"
            >
              <section
                class="top-box"
              >
                <h5
                  class="MuiTypography-root heading Typography MuiTypography-h5"
                >
                  Call to action
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root icon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
                    />
                  </svg>
                </h5>
              </section>
              <div
                class="image Media gatsby-image-wrapper"
                style="overflow: hidden;"
              >
                <div
                  aria-hidden="true"
                  style="width: 100%; padding-bottom: 100%;"
                />
                <picture>
                  
                  <img
                    alt="Lorem ipsum"
                    draggable="false"
                    loading="lazy"
                    sizes=""
                    src=""
                    srcset=""
                    style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 0; transition: opacity 500ms; font-family: \\"object-fit: cover; object-position: center\\";"
                  />
                </picture>
                <noscript>
                  &lt;picture&gt;&lt;source srcset="" /&gt;&lt;img loading="lazy" src="" alt="Lorem ipsum" style="position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
                </noscript>
              </div>
            </div>
            <span
              class="MuiTouchRipple-root"
            />
          </a>
        </div>
        <div
          class="MuiGrid-root item MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-5 MuiGrid-grid-lg-3"
        >
          <div
            class="Card"
          >
            <div
              class="MuiPaper-root MuiCard-root body MuiPaper-elevation1 MuiPaper-rounded"
            >
              <a
                class="MuiButtonBase-root top-box"
                href="/uk/some-page"
              >
                <h5
                  class="MuiTypography-root heading Typography MuiTypography-h5"
                >
                  Card with Video
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root icon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
                    />
                  </svg>
                </h5>
                <span
                  class="MuiTouchRipple-root"
                />
              </a>
              <div
                class="image Media"
              >
                <div
                  class="YoutubeVideo"
                  style="background-image: url(link-to-page.png);"
                >
                  <span
                    class="AlternativeContent"
                  >
                    video label
                  </span>
                  <button
                    class="MuiButtonBase-root MuiIconButton-root IconButton IconButton--medium IconButton--contained play-button"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M8 5v14l11-7z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
